tests/test_utils/test_config.py

```diff
@@ -314,3 +316,237 @@ def test_validate_config_floats(
             validate_config(dummy_config)
+
+
+class TestValidateFreezingConfigSettings(object):
+    """Tests for validate_freezing_config_settings."""
+
+    def test_validate_freezing_config_settings_no_errors(self):
+        """Test validate_freezing_config_settings with no errors."""
+        user_config = {
+            "global": {
+                "run_first_snapshot_of_results": False,
+                "run_frozen_data": True,
+                "load_updated_snapshot_for_comparison": False,
+                "run_updates_and_freeze": True,
+            },
+            "hdfs_paths": {
+                "frozen_snapshot_path": "/path/to/frozen_snapshot",
+                "frozen_data_staged_path": "/path/to/frozen_data_staged",
+                "secondary_snapshot_path": "/path/to/secondary_snapshot",
+                "freezing_additions_path": "/path/to/freezing_adds",
+                "freezing_amendments_path": "/path/to/freezing_amends",
+            },
+        }
+        validate_freezing_config_settings(user_config)
+
+    def test_validate_freezing_config_settings_run_first_snapshot_and_run_frozen_data(self):
+        """Test validate_freezing_config_settings with run_first_snapshot_of_results and run_frozen_data both True."""
+        user_config = {
+            "global": {
+                "run_first_snapshot_of_results": True,
+                "run_frozen_data": True,
+                "load_updated_snapshot_for_comparison": False,
+                "run_updates_and_freeze": True,
+            },
+            "hdfs_paths": {
+                "frozen_snapshot_path": "/path/to/frozen_snapshot",
+                "frozen_data_staged_path": "/path/to/frozen_data_staged",
+                "secondary_snapshot_path": "/path/to/secondary_snapshot",
+                "freezing_additions_path": "/path/to/freezing_adds",
+                "freezing_amendments_path": "/path/to/freezing_amends",
+            },
+        }
+        with pytest.raises(ValueError):
+            validate_freezing_config_settings(user_config)
+
+    def test_validate_freezing_config_settings_run_first_snapshot_without_frozen_snapshot_path(self):
+        """Test validate_freezing_config_settings with run_first_snapshot_of_results True but no frozen_snapshot_path."""
+        user_config = {
+            "global": {
+                "run_first_snapshot_of_results": True,
+                "run_frozen_data": False,
+                "load_updated_snapshot_for_comparison": False,
+                "run_updates_and_freeze": True,
+            },
+            "hdfs_paths": {
+                "frozen_snapshot_path": None,
+                "frozen_data_staged_path": "/path/to/frozen_data_staged",
+                "secondary_snapshot_path": "/path/to/secondary_snapshot",
+                "freezing_additions_path": "/path/to/freezing_adds",
+                "freezing_amendments_path": "/path/to/freezing_amends",
+            },
+        }
+        with pytest.raises(ValueError):
```

Comment:
a match here would be useful to ensure the correct error is raised. this function can raise multiple value errors